Drakhtar/Battalion.cpp

```diff
@@ -1,14 +1,41 @@
 // Copyright 2019 the Drakhtar authors. All rights reserved. MIT license.
 
 #include "Battalion.h"
+#include "Game.h"
+
+
+Battalion::Battalion(Texture * texture, Box * box, int attack, int health,
+int speed, int attackRange, int moveRange, int battalionSize)
+    : Unit(texture, box, attack, health, speed, attackRange, moveRange),
+    battalionSize_(battalionSize)
+    {
+    health_ = Unit::getMaxHealth() * battalionSize_;
+
+    SDL_Color textColor;
+    textColor.r = 255;
+    textColor.g = 255;
+    textColor.b = 255;
+    textColor.a = 255;
+    healthText->setText("Salud: " + to_string(this->getHealth()));
+
+    sizeText_ =
+    new Text(Game::getInstance()->getRenderer(), FontManager::get("Retron2000"),
+    { box_->getRect().x + box_->getRect().w / 2 , box_->getRect().y + box_->getRect().h * 4 / 5 },
+    textColor, "Units: " + to_string(battalionSize_), box_->getRect().w * 2);
+}
 
 Battalion::~Battalion()
 {
+    if (sizeText_ != nullptr) {
+        delete sizeText_;
+    }
+    sizeText_ = nullptr;
```

Comment:
Move this line inside the if, we don't need the assignment if it's already null.